libtraceevent/parse-filter.c

```diff
@@ -1624,9 +1624,11 @@ int pevent_filter_clear_trivial(struct event_filter *filter,
 		case FILTER_TRIVIAL_FALSE:
 			if (filter_type->filter->boolean.value)
 				continue;
+			/* fall through */
 		case FILTER_TRIVIAL_TRUE:
 			if (!filter_type->filter->boolean.value)
 				continue;
+			/* fall through */
```

Comment:
I understood it, thank you.
I thought same thing about the case `FILTER_TRIVIAL_FALSE:`.
Just use `continue;` like below ? as another PR 
```
diff --git a/libtraceevent/parse-filter.c b/libtraceevent/parse-filter.c
index 21c92fb..60ad0a8 100644
--- a/libtraceevent/parse-filter.c
+++ b/libtraceevent/parse-filter.c
@@ -1622,11 +1622,11 @@ int pevent_filter_clear_trivial(struct event_filter *filter,
                         continue;
                 switch (type) {
                 case FILTER_TRIVIAL_FALSE:
-                        if (filter_type->filter->boolean.value)
-                                continue;
+                        continue;
                 case FILTER_TRIVIAL_TRUE:
                         if (!filter_type->filter->boolean.value)
                                 continue;
+                        break;
                 default:
                         break;
                 }

```